xbmc/filesystem/CurlFile.cpp

```diff
@@ -540,7 +540,7 @@ void CCurlFile::SetCommonOptions(CReadState* state)
   if (m_postdataset)
   {
     g_curlInterface.easy_setopt(h, CURLOPT_POST, 1 );
-    g_curlInterface.easy_setopt(h, CURLOPT_POSTFIELDSIZE, m_postdata.length());
+    g_curlInterface.easy_setopt(h, CURLOPT_POSTFIELDSIZE, m_postdata.size());
```

Comment:
I could provide some binary data where length() is completely different to size().
But - I realy know this from experiences at work with UTF8.

Need the examle? Or trust me?